cosmos/precompile/bank/bank.go

```diff
@@ -312,6 +316,61 @@ func (c *Contract) Send(
 	return err == nil, err
 }
 
+// Allowance implements `allowance(address,string)` method.
+func (c *Contract) Allowance(ctx context.Context, spenderAddress common.Address, denom string) (*big.Int, error) {
+	caller, err := cosmlib.StringFromEthAddress(
+		c.addressCodec, vm.UnwrapPolarContext(ctx).MsgSender(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	spender, err := cosmlib.StringFromEthAddress(c.addressCodec, spenderAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.authzQuerier.Grants(
```

Comment:
ask; Is there any way to abstract the query like message routing service?

Reply:
we can use just `msgRouter`. but I thought that it could be annoying to parse the response.

Reply:
Ah i just found the GRPCQueryRouter in baseapp. So can we use this?

Reply:
I guess yes. but it's same that it's hard to parse the result of specific type from the response. right?

Reply:
I'll merge it to continue my work. but let's continue to discuss and we could improve in another PR.